### PR TITLE
add integration tests for status condition Failed type usecases

### PR DESCRIPTION
**Pull request by sankarpn**
Adding integration tests for status condition failed type usecases. 

https://build.weblogick8s.org:8443/job/weblogic-kubernetes-operator-kind-new/7414/

### CODE REVIEW

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItDiagnosticsFailedCondition.java [hunk 1]
```diff
@@ -0,0 +1,668 @@
...
+
+    } finally {
+      deleteDomainResource(domainUid, domainNamespace);
+      deleteConfigMap(badModelFileCm, domainNamespace);
+    }
+  }
+
+  /**
+   * Test domain status condition with replicas set to more than maximum size of the WebLogic cluster created.
+   * generated: type: Completed, status: true type: Available, status: false type: Failed status: true
```
Comment (anpanigr) [inline integration-tests/src/test/java/oracle/weblogic/kubernetes/ItDiagnosticsFailedCondition.java:189]: Maintain the same description format as previous test

 type: Completed, status: true
 type: Available, status: false
 type: Failed, status: true

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItDiagnosticsFailedCondition.java [hunk 2]
```diff
@@ -0,0 +1,668 @@
...
+    // create encryption secret
+    logger.info("Create encryption secret");
+    encryptionSecretName = "encryptionsecret";
+    createSecretWithUsernamePassword(encryptionSecretName, domainNamespace,
+            "weblogicenc", "weblogicenc");
+  }
+
+  /**
+   * Test domain status condition with a bad model file.
+   * Verify the following conditions are generated after a introspector failure.
```
Comment (anpanigr) [inline integration-tests/src/test/java/oracle/weblogic/kubernetes/ItDiagnosticsFailedCondition.java:151]: Verify the following conditions are generated after a introspector failure. --> Verify the following conditions are generated in an order after a introspector failure.

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItDiagnosticsFailedCondition.java [hunk 3]
```diff
@@ -0,0 +1,668 @@
...
+    }
+  }
+
+  /**
+   * Test domain status failed condition with incorrect image pull secret. Verify the following conditions are
+   * generated: type: Completed, status: true type: Available, status: false type: Failed status: true
+   */
+  @Test
+  @DisplayName("Test domain status condition with incorrect image pull secret")
+  void testIncorrectImagePullSecret() {
```
Comment (anpanigr) [inline integration-tests/src/test/java/oracle/weblogic/kubernetes/ItDiagnosticsFailedCondition.java:268]: Here we are generating two Error conditions (a) bad image name (b) bad image pull secret.  We should create one error condition per usecase. May be we should have different test case that  generate multiple error conditions

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItDiagnosticsFailedCondition.java [hunk 4]
```diff
@@ -0,0 +1,668 @@
...
+  }
+
+  /**
+   * Test domain status failed condition with non-existent admin secret. Verify the following conditions are
+   * generated: type: Completed, status: true type: Available, status: false type: Failed status: true
+   */
+  @Test
+  @DisplayName("Test domain status condition with non-existent admin secret")
+  void testNonexistentAdminSecret() {
+    String image = TestConstants.MII_BASIC_IMAGE_NAME + ":" + TestConstants.MII_BASIC_IMAGE_TAG;
```
Comment (anpanigr) [inline integration-tests/src/test/java/oracle/weblogic/kubernetes/ItDiagnosticsFailedCondition.java:365]: TestConstants.MII_BASIC_IMAGE_TAG-->MII_BASIC_IMAGE_TAG
TestConstants.MII_BASIC_IMAGE_NAME --> MII_BASIC_IMAGE_NAME

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItDiagnosticsFailedCondition.java [hunk 5]
```diff
@@ -0,0 +1,668 @@
...
+  }
+
+
+  /**
+   * Test domain status condition with serverStartPolicy set to IF_NEEDED. Verify the following conditions are
+   * generated: type: Completed, status: true type: Available, status: true Verify no Failed type condition generated.
+   */
+  @Test
+  @DisplayName("Test domain status condition with serverStartPolicy set to IF_NEEDED")
+  void testMSBootFailureStatus() {
```
Comment (anpanigr) [inline integration-tests/src/test/java/oracle/weblogic/kubernetes/ItDiagnosticsFailedCondition.java:451]: The description does not mach the test name.

File: integration-tests/src/test/resources/wdt-models/bad-model-file.yaml [hunk 6]
```diff
@@ -0,0 +1,19 @@
+# Copyright (c) 2020, 2021, Oracle and/or its affiliates.
```
Comment (anpanigr) [inline integration-tests/src/test/resources/wdt-models/bad-model-file.yaml:1]: copyright (c) 2020, 2021, Oracle and/or its affiliates. --> copyright (c)  2021, Oracle and/or its affiliates.

**Review by anpanigr — COMMENTED** [review]
Need to verify the status message in case of Failed ( true )
If multiple failed conditioned are generated, do we see the messages related all failed condition

We can disable the test which is not working due to existing BUG and update the BUG with reproducer.  We can run the  integration tests with disabled tests. We did the same with Positive diagnostic usecases.

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItDiagnosticsFailedCondition.java [hunk 7]
```diff
@@ -0,0 +1,668 @@
...
+    } finally {
+      deleteDomainResource(domainUid, domainNamespace);
+    }
+  }
+
+  /**
+   * Cleanup the persistent volume and persistent volume claim used by the test.
+   */
+  @AfterAll
+  public static void tearDown() {
```
Comment (xiancao) [inline integration-tests/src/test/java/oracle/weblogic/kubernetes/ItDiagnosticsFailedCondition.java:528]: I think the pv and pvc should be cleanup by the framework.

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItDiagnosticsFailedCondition.java [hunk 8]
```diff
@@ -0,0 +1,668 @@
...
+   * Test domain status condition with a bad model file.
+   * Verify the following conditions are generated after a introspector failure.
+   * type: Completed, status: true
+   * type: Available, status: false
+   * type: Failed, status: true
+   */
+  @Test
+  @DisplayName("Test domain status condition with bad model file")
+  void testBadModelFileStatus() {
+    // build a build with empty WebLogic domain
```
Comment (xiancao) [inline integration-tests/src/test/java/oracle/weblogic/kubernetes/ItDiagnosticsFailedCondition.java:159]: change to -> // build an image with empty WebLogic domain

**Review by xiancao — CHANGES_REQUESTED** [review]

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItDiagnosticsFailedCondition.java (hunk 8 above)
Comment (sankarpn) [reply integration-tests/src/test/java/oracle/weblogic/kubernetes/ItDiagnosticsFailedCondition.java]: done

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItDiagnosticsFailedCondition.java (hunk 7 above)
Comment (sankarpn) [reply integration-tests/src/test/java/oracle/weblogic/kubernetes/ItDiagnosticsFailedCondition.java]: removed tearDown

File: integration-tests/src/test/resources/wdt-models/bad-model-file.yaml (hunk 6 above)
Comment (sankarpn) [reply integration-tests/src/test/resources/wdt-models/bad-model-file.yaml]: changed

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItDiagnosticsFailedCondition.java (hunk 5 above)
Comment (sankarpn) [reply integration-tests/src/test/java/oracle/weblogic/kubernetes/ItDiagnosticsFailedCondition.java:525]: changed

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItDiagnosticsFailedCondition.java (hunk 4 above)
Comment (sankarpn) [reply integration-tests/src/test/java/oracle/weblogic/kubernetes/ItDiagnosticsFailedCondition.java:414]: fixed

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItDiagnosticsFailedCondition.java (hunk 3 above)
Comment (sankarpn) [reply integration-tests/src/test/java/oracle/weblogic/kubernetes/ItDiagnosticsFailedCondition.java:295]: that was a mistake, using the correct image now

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItDiagnosticsFailedCondition.java (hunk 2 above)
Comment (sankarpn) [reply integration-tests/src/test/java/oracle/weblogic/kubernetes/ItDiagnosticsFailedCondition.java]: changed

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItDiagnosticsFailedCondition.java (hunk 1 above)
Comment (sankarpn) [reply integration-tests/src/test/java/oracle/weblogic/kubernetes/ItDiagnosticsFailedCondition.java]: following same format for all tests

**Comment by sankarpn** [comment]
> Need to verify the status message in case of Failed ( true ) If multiple failed conditioned are generated, do we see the messages related all failed condition
> 
> We can disable the test which is not working due to existing BUG and update the BUG with reproducer. We can run the integration tests with disabled tests. We did the same with Positive diagnostic usecases.

There are 3 tests failing with bugs.

These 2 tests missing Failed and Complete conditions
[ERROR]   ItDiagnosticsFailedCondition.testImagePullSecretDoesnotExist:275->checkStatus:721 » Timeout
[ERROR]   ItDiagnosticsFailedCondition.testNonexistentAdminSecret:425->checkStatus:721 » Timeout

This test is missing Failed condition
[ERROR]   ItDiagnosticsFailedCondition.testReplicasTooHigh:207->checkStatus:713 » Timeout

**Review by anpanigr — APPROVED** [review]
lgtm

**Review by xiancao — APPROVED** [review]
lgtm

**Pull request merged**